packages/smooth_app/lib/data_models/up_to_date_manager.dart

```diff
@@ -0,0 +1,38 @@
+import 'package:openfoodfacts/openfoodfacts.dart';
+import 'package:smooth_app/background/background_task_manager.dart';
+import 'package:smooth_app/database/local_database.dart';
+
+/// Provides the most up-to-date local product data for a StatefulWidget.
+///
+/// Typically we have
+/// * a product from the database (downloaded from the server)
+/// * potentially pending changes to apply on top while they're being uploaded
+///
+/// With this class
+/// * we get the most up-to-date local product data
+/// * we re-launch the task manager if relevant
+/// * we track the barcodes currently "opened" by the app
+class UpToDateManager {
```

Comment:
Why don't use a `mixin` here?
As it's specifically design for a `StatefullWidget`, it would simplify the integration

Otherwise, looks OK 👍

Reply:
@g123k Thank you for the suggestion! I edited the code in the latest push: my first experience with `mixin`.

Not 100% convinced in that particular case, specifically because we somehow need a constructor - replaced by an "init" method as `mixin` doesn't accept constructors.
In the end, it looks a bit better than before.

Reply:
I clearly prefer this implementation, even if indeed the number of lines is similar.

(I am also very sorry, I receive random notifications from GitHub, and I totally missed your answer)